R/draw-dm.R

```diff
@@ -146,6 +146,47 @@ dm_get_all_columns <- function(x) {
 #'   dm_draw()
 #' @export
 dm_set_colors <- function(dm, ...) {
+  avail_cols <- dm_get_available_colors()$dm
+  quos <- enquos(...)
+  # If old fall-through syntax is being used:
+  if (!all(names(quos) %in% avail_cols) &&
+    all(names(quos) %in% src_tbls(dm))) {
+    warning(paste0(
+      "When setting colors with `dm_set_colors()`, ",
+      "the syntax is now as follows: `blue = table_name1, ",
+      "orange = table_name2` etc. (tidyselect supported) ",
+      "Forwarding the function call ",
+      "to the old implementation of `dm_set_colors()`."
+    ))
+    return(dm_set_colors2(dm, !!!quos))
+  }
+
+  avail_tables <- src_tbls(dm)
+  selected_tables <- tidyselect::vars_select(avail_tables, ...) %>%
+    # names will get integer suffixes if duplicated ("blue1", "blue2", ...)
+    set_names(gsub("[0-9]*", "", names(.)))
+  sel_colors <- names(selected_tables)
+  if (!all(sel_colors %in% colors$dm)) {
+    abort_wrong_color(paste0("`", colors$dm, "` ", colors$nb))
+  }
+
+  display_df <- tibble(
+    table = selected_tables,
+    new_display = colors$datamodelr[match(sel_colors, colors$dm)]
```

Comment:
Can we drop support for datamodelr colors, and instead support a much wider range of colors, and also arbitrary RGB colors?

Reply:
Now there is a much larger choice of colors (`mauve` is still missing though, sorry ;) ). Basically all hex colors can be used for setting the header color or alternatively you can use the color names given by `grDevices::colors()`.

I still left the {datamodelr} colortables untouched, cause the legacy `cdm_set_colors()` still uses them. But the implementation for the new function is independent of them.